### PR TITLE
Fixed bug with JSON output

### PR DESCRIPTION
**Pull request by RackerWilliams**
Fixed a bug where multi-value attributes were not always returned as
an array when generating JSON output.

### CODE REVIEW

File: core/src/test/resources/tests/mapping-tests/adfs-faws-ext1/maps/adfs-s-sj.yaml [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+    - regex: false
+      name: http://schemas.xmlsoap.org/claims/Group
+      anyOneOf: Contractor
+    local:
+      faws:
+        nones:
+          value:
+          - 12285/AWSPolicy
+          - 38839/AWSPolicy
+          multiValue: true
```
Comment (wdschei) [inline core/src/test/resources/tests/mapping-tests/adfs-faws-ext1/maps/adfs-s-sj.yaml]: I know this is just a test, but if the description says this is an "example". Should the `value`'s be different for each?

File: core/src/test/resources/tests/mapping-tests/adfs-faws-ext2/maps/faws-sj.yaml [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+        991049284483:
+          value:
+            - fanatical_aws:admin
+            - AdminstratorAccess
+          multiValue: true
+        042423532529:
+          value:
+            - fanatical_aws:observer
+            - RackspaceReadOnly
+          multiValue: true
```
Comment (wdschei) [inline core/src/test/resources/tests/mapping-tests/adfs-faws-ext2/maps/faws-sj.yaml]: Like this one shows.

File: core/src/test/resources/tests/mapping-tests/adfs-faws-ext1/maps/adfs-s-sj.yaml (hunk 1 above)
Comment (RackerWilliams) [reply core/src/test/resources/tests/mapping-tests/adfs-faws-ext1/maps/adfs-s-sj.yaml]: Well tests may turn into examples in the docs -- not sure this one will though.  I don't follow, the values are different -- no?

File: core/src/test/resources/tests/mapping-tests/adfs-faws-ext2/maps/faws-sj.yaml (hunk 2 above)
Comment (RackerWilliams) [reply core/src/test/resources/tests/mapping-tests/adfs-faws-ext2/maps/faws-sj.yaml]: Totally different test from the first one.

File: core/src/test/resources/tests/mapping-tests/adfs-faws-ext1/maps/adfs-s-sj.yaml (hunk 1 above)
Comment (wdschei) [reply core/src/test/resources/tests/mapping-tests/adfs-faws-ext1/maps/adfs-s-sj.yaml]: The value for `faws:admins`, `faws:observers`, & `faws:nones` is always `12285/AWSPolicy` and `38839/AWSPolicy`.
Or am I missing something?

File: core/src/test/resources/tests/mapping-tests/adfs-faws-ext2/maps/faws-sj.yaml (hunk 2 above)
Comment (wdschei) [reply core/src/test/resources/tests/mapping-tests/adfs-faws-ext2/maps/faws-sj.yaml]: I get this is a different test case, but `faws:991049284483` has the values `fanatical_aws:admin` and `AdminstratorAccess` while `faws:042423532529` has the values `fanatical_aws:observer` and `RackspaceReadOnly`. That is what I was referring too.

File: core/src/test/resources/tests/mapping-tests/adfs-faws-ext1/maps/adfs-single-rule-sj.yaml [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+  - remote:
+    - path: if (mapping:get-attributes('http://schemas.xmlsoap.org/claims/Group')='Admin')
+        then ('12285/AWSPolicy','38839/AWSPolicy') else ()
+      multiValue: true
+    - path: if (not(mapping:get-attributes('http://schemas.xmlsoap.org/claims/Group')='Contractor'))
+        then ('12285/AWSPolicy','38839/AWSPolicy') else ()
+      multiValue: true
+    - path: if (mapping:get-attributes('http://schemas.xmlsoap.org/claims/Group')='Contractor')
+        then ('12285/AWSPolicy','38839/AWSPolicy') else ()
+      multiValue: true
```
Comment (wdschei) [inline core/src/test/resources/tests/mapping-tests/adfs-faws-ext1/maps/adfs-single-rule-sj.yaml]: This one has if the group is `Admin` then `12285/AWSPolicy` and `38839/AWSPolicy`.
This is also the result if the group is not `Contractor` or if it is `Contractor`.
Again, am I missing something?

Comment (RackerWilliams) [reply core/src/test/resources/tests/mapping-tests/adfs-faws-ext1/maps/adfs-single-rule-sj.yaml]: The test is correct, though the policy language is limited so it's a bit verbose.

`12285/AWSPolicy` and `38839/AWSPolicy`  are the list of accounts.  FAWS keeps a list of accounts your are admin on,  accounts you are observers on and accounts where you have no permission.

Based on what groups you have, we place the account in the appropriate bucket.  The assertions check to make sure they land in the right place.

File: core/src/test/resources/tests/mapping-tests/adfs-faws-ext1/maps/adfs-s-sj.yaml (hunk 1 above)
Comment (RackerWilliams) [reply core/src/test/resources/tests/mapping-tests/adfs-faws-ext1/maps/adfs-s-sj.yaml]: ...no not always.   Depending on the group you have you'll have those accounts in `faws:admin`, `faws:observers`, or `faws;nones`.

File: core/src/test/resources/tests/mapping-tests/adfs-faws-ext2/maps/faws-sj.yaml (hunk 2 above)
Comment (RackerWilliams) [reply core/src/test/resources/tests/mapping-tests/adfs-faws-ext2/maps/faws-sj.yaml]: That's fine. this was a test of a different model for extension attributes. Essentially  what this is saying is that if you are a member of test_group_2 you should have  the ability to add aws accounts,  and the groups `fanatical_aws:admin` and `AdminstratorAccess` on account 991049284483 and the groups `fanatical_aws:observer` and `RackspaceReadOnly` on account 042423532529.  Make sense?

File: core/src/test/resources/tests/mapping-tests/adfs-faws-ext1/maps/adfs-s-sj.yaml (hunk 1 above)
Comment (wdschei) [reply core/src/test/resources/tests/mapping-tests/adfs-faws-ext1/maps/adfs-s-sj.yaml]: :+1:

File: core/src/test/resources/tests/mapping-tests/adfs-faws-ext2/maps/faws-sj.yaml (hunk 2 above)
Comment (wdschei) [reply core/src/test/resources/tests/mapping-tests/adfs-faws-ext2/maps/faws-sj.yaml]: :+1:

File: core/src/test/resources/tests/mapping-tests/adfs-faws-ext1/maps/adfs-single-rule-sj.yaml (hunk 3 above)
Comment (wdschei) [reply core/src/test/resources/tests/mapping-tests/adfs-faws-ext1/maps/adfs-single-rule-sj.yaml]: :+1:

**Pull request merged**